### PR TITLE
Add userspace config.h handling to build script

### PR DESCRIPTION
**Pull request by drashna**
The reason for this is that the user.h file doesn't get called in such a way that will modify QMK settings.  (ie, my settings were not being honored).    

Additionally, if I included the file in the config.h file, it would error out.
* with `#include "quantum.h"` would attempt to redefine everything in them, and error.
*without would not define `SAFE_RANGE` and stuff like "bool". 

So this allows for you to use a config.h file in the userspace folder that is processed like the keyboard/keymap file.   Thus allowing for more consistent and global handling of settings.  

TL;DR: `<name>.h` is for user specific stuff, and `config.h` is for globally configuring keyboard settings

Caveat: it would be best if I could figure out how to get `<name>.h` added to all `keymap.c` files, but that can wait.

### CODE REVIEW

**Comment by drashna** [comment]
Instead of a "big" change, I can simply add the specific code to the `rules.mk` in my userspace, and it works identically. 

```
ifneq ("$(wildcard users/$(KEYMAP)/config.h)","")
    CONFIG_H += users/$(KEYMAP)/config.h
endif
```

So .... should this be added globally, or just left as is, and for people to do this themselves?
